src/common/base/CollectNSucceeded.inl

```diff
@@ -45,11 +47,11 @@ folly::Future<SucceededResultList<FutureIter>> collectNSucceeded(
     // for each succeeded Future, add to the result list, until
     // we have required number of futures, at which point we fulfil
     // the promise with the result list
-    for (; first != last; ++first) {
-        first->setCallback_([n, ctx] (
+    for (size_t index = 0; first != last; ++first, ++index) {
+        first->setCallback_([n, ctx, index] (
                 folly::Try<FutureReturnType<FutureIter>>&& t) {
             if (!ctx->promise.isFulfilled()) {
-                if (!t.hasException() && ctx->eval(t.value())) {
+                if (!t.hasException() && ctx->eval(index, t.value())) {
```

Comment:
maybe we can not call emplace_back to save the result, because we will check the response in RaftPart::processAppendLogResponses, and in RaftPart::processAppendLogResponses, we strong dependence the index to judge the host is learner or not.

Reply:
Good catch.